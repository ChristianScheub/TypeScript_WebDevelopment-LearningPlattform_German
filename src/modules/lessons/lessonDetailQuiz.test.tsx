import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import QuizComponent from './lessonDetailQuiz';  

describe('QuizComponent', () => {
  const mockCheckQuizAnswer = jest.fn();
  const formRef = React.createRef<HTMLFormElement>();

  const props = {
    lesson: {
      quizText: 'Was ist die Hauptstadt von Frankreich?',
      quizSolution: 'Paris',
      quizOptions: 'Berlin;Paris;Rom'
    },
    showCongratulationsOverlay: null,
    checkQuizAnswer: mockCheckQuizAnswer,
    formRef: formRef,
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<QuizComponent {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render options correctly', () => {
    render(<QuizComponent {...props} />);
    expect(screen.getByLabelText('Berlin')).toBeInTheDocument();
    expect(screen.getByLabelText('Paris')).toBeInTheDocument();
    expect(screen.getByLabelText('Rom')).toBeInTheDocument();
  });

  it('should call checkQuizAnswer when button is clicked', () => {
    render(<QuizComponent {...props} />);
    fireEvent.click(screen.getByText('Überprüfen'));
    expect(mockCheckQuizAnswer).toHaveBeenCalledTimes(1);
  });
});
