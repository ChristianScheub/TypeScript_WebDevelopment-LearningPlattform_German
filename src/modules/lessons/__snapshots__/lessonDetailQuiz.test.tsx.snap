// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizComponent should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-title h5"
  >
    Was ist die Hauptstadt von Frankreich?
  </div>
  <form
    class=""
  >
    <div
      class="mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="option-0"
          name="quizOption"
          type="radio"
          value="Berlin"
        />
        <label
          class="form-check-label"
          for="option-0"
          title=""
        >
          Berlin
        </label>
      </div>
    </div>
    <div
      class="mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="option-1"
          name="quizOption"
          type="radio"
          value="Paris"
        />
        <label
          class="form-check-label"
          for="option-1"
          title=""
        >
          Paris
        </label>
      </div>
    </div>
    <div
      class="mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="option-2"
          name="quizOption"
          type="radio"
          value="Rom"
        />
        <label
          class="form-check-label"
          for="option-2"
          title=""
        >
          Rom
        </label>
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Überprüfen
    </button>
  </form>
  <br />
  <br />
</DocumentFragment>
`;
